src/control/provider/system/system_linux.go

```diff
@@ -165,3 +175,157 @@ func (s LinuxProvider) GetfsUsage(target string) (uint64, uint64, error) {
 
 	return frSize * stBuf.Blocks, frSize * stBuf.Bavail, nil
 }
+
+func (s LinuxProvider) checkDevice(device string) error {
+	st, err := s.Stat(device)
+	if err != nil {
+		return errors.Wrapf(err, "stat failed on %s", device)
+	}
+
+	if st.Mode()&os.ModeDevice == 0 {
+		return errors.Errorf("%s is not a device file", device)
+	}
+
+	return nil
+}
+
+func getDistroArgs() []string {
+	distro := GetDistribution()
+
+	// use stride option for SCM interleaved mode
+	// disable lazy initialization (hurts perf)
+	// discard is not needed/supported on SCM
+	opts := "stride=512,lazy_itable_init=0,lazy_journal_init=0,nodiscard"
```

Comment:
Yeah, I think `Mkfs()` will need to be aware of the underlying device type and its intended usage in order to generate the correct set of options. I think it's fine to land this as-is because there is another PR coming after this one that adds support for managing a new mountpoint for control plane data.

Reply:
OK thanks for the info :)